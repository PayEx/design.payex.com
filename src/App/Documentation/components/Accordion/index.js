import React, { useEffect } from "react";

import { ComponentPreview, DocContainer } from "#";
import accordion from "$/px-script/main/accordion";
import AccordionComponent from "@/Accordion";

const items = [
    {
        title: "Foo",
        content: <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada.
            Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus.
            Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat.
            Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
        </p>
    },
    {
        title: "Bar",
        content: <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada.
            Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus.
            Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat.
            Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
        </p>
    },
    {
        title: "Baz",
        content: <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada.
            Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus.
            Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat.
            Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
        </p>
    }
];

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Accordion.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <AccordionComponent items={items} />
        </ComponentPreview>
    </>
);

const Accordion = () => {
    useEffect(() => accordion.init());

    return (
        <DocContainer docToc>
            <p className="lead">
                Accordion
            </p>
            <Overview />
        </DocContainer>
    );
};

export default Accordion;

/* For testing */
export { Overview };
