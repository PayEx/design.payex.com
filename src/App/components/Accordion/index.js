import React from "react";

const Accordion = ({ items }) => (
    <div className="accordion">
        {items ? items.map(({ title, content }, i) => (
            <div key={i} className="accordion-item">
                <h5>{title}</h5>
                <div className="accordion-content">
                    {content}
                </div>
            </div>
        )) : null}
    </div>
);

export default Accordion;
