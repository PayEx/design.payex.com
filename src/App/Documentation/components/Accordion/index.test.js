import React from "react";
import { shallow } from "enzyme";

import Accordion, { Overview } from "./index";

describe("Components: Accordion", () => {
    it("is defined", () => {
        expect(Accordion).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Accordion />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
