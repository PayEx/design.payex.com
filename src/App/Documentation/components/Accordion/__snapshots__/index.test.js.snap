// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Accordion Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Accordion.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Accordion
      items={
        Array [
          Object {
            "content": <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada. Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus. Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat. Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
            </p>,
            "title": "Foo",
          },
          Object {
            "content": <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada. Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus. Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat. Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
            </p>,
            "title": "Bar",
          },
          Object {
            "content": <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer tristique scelerisque blandit. Vestibulum venenatis elit non nunc pulvinar, vitae tincidunt quam malesuada. Etiam orci dui, ultricies consectetur venenatis sit amet, ullamcorper quis metus. Aliquam eget consequat erat. Praesent arcu tortor, eleifend ac lorem eu, ullamcorper tristique purus. Phasellus felis mi, malesuada et egestas eget, efficitur at magna. Fusce euismod, elit tempor vulputate mollis, diam eros venenatis libero, eget ultricies diam velit at erat. Morbi mattis scelerisque augue, ut volutpat lectus sodales iaculis.
            </p>,
            "title": "Baz",
          },
        ]
      }
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Accordion renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Accordion
  </p>
  <Overview />
</DocContainer>
`;
